src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -349,16 +328,22 @@ namespace winrt::TerminalApp::implementation
 
             e.Handled(true);
         }
-        else if (key == VirtualKey::Back)
+        else if (key == VirtualKey::Back && _searchBox().Text().empty() && _lastFilterTextWasEmpty && _currentMode == CommandPaletteMode::ActionMode)
         {
             // If the last filter text was empty, and we're backspacing from
             // that state, then the user "backspaced" the virtual '>' we're
             // using as the action mode indicator. Switch into commandline mode.
-            if (_searchBox().Text().empty() && _lastFilterTextWasEmpty && _currentMode == CommandPaletteMode::ActionMode)
-            {
-                _switchToMode(CommandPaletteMode::CommandlineMode);
-            }
-
+            _switchToMode(CommandPaletteMode::CommandlineMode);
+            e.Handled(true);
+        }
+        else if (key == VirtualKey::C && ctrlDown)
+        {
+            _searchBox().CopySelectionToClipboard();
+            e.Handled(true);
+        }
+        else if (key == VirtualKey::V && ctrlDown)
+        {
+            _searchBox().PasteFromClipboard();
             e.Handled(true);
         }
```

Comment:
Should we also hardcode some other text box default bindings like <kbd>ctrl+a</kbd>, <kbd>ctrl+left/right</kbd>, <kbd>ctrl+x</kbd>, and <kbd>home/end</kbd>? Or what makes <kbd>ctrl+c</kbd> and <kbd>ctrl+v</kbd> so special here?

Reply:
Isn't there a way to ask the text box to handle the event first, and if it can't then we handle it? Or how does the order of handling work?

Reply:
Yes. Probably. I handled copy and paste explicitly as right now it is the same default binding for pasting in the terminal. Which means that the current behavior upon ctrl+v is to paste in the terminal.